SCMI SCMI_SYSTEM_MSG_POWER_STATE_SET not defined in scmi/reboot.c

### Describe the bug

The PR https://github.com/zephyrproject-rtos/zephyr/pull/109215 unify parameter name about scmi protocol, but not about scmi/reboot.c which continue use SCMI_SYSTEM_MSG_POWER_STATE_SET, so induced build fail issue in local and test twister ci

### Regression

- [ ] This is a regression.

### Steps to reproduce

west build -p -b imx95_evk/mimx9596/m7/ddr tests/arch/arm/arm_runtime_nmi -T arch.arm.interrupt.nmi
CI check fail: https://github.com/zephyrproject-rtos/zephyr/actions/runs/26275958356/job/77340965396?pr=103541

### Relevant log output

```shell

```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

This issue is also reproduced in NXP internal CI (Jenkins `bitbucket_build_platform` build #9656) on platforms `imx943_evk/mimx94398/m33` and `imx943_evk/mimx94398/m33/ddr`.

**Zephyr version:** `v4.4.0-3337-g07e34e68d340`

**Build command:**
```
scripts/twister -p imx943_evk/mimx94398/m33 -j 12 -T tests -T samples --build-only --inline-logs -M all
```

**Error:**
```
drivers/firmware/scmi/reboot.c:24: error: 'SCMI_SYSTEM_MSG_POWER_STATE_SET' undeclared
(first use in this function); did you mean 'SCMI_SYSTEM_POWER_STATE_SUSPEND'?
  ret = scmi_system_protocol_message_attributes(SCMI_SYSTEM_MSG_POWER_STATE_SET,
```

**Impact:** 117 build failures across `tests/` and `samples/` on `imx943_evk/mimx94398/m33` — all caused by the same root cause in `drivers/firmware/scmi/reboot.c`.

Looking forward to PR #109737 being merged to resolve this.

--- a/drivers/firmware/scmi/reboot.c
+++ b/drivers/firmware/scmi/reboot.c
@@ -15,24 +15,11 @@ static int scmi_reboot_handler(int type)
 {
 	struct scmi_system_power_state_config cfg;
 	int ret;
-	uint32_t mesg_attr;
 
 	cfg.flags = SCMI_SYSTEM_POWER_FLAG_FORCEFUL;
 
 	switch (type) {
 	case SYS_REBOOT_WARM:
-		ret = scmi_system_protocol_message_attributes(SCMI_SYSTEM_MSG_POWER_STATE_SET,
-							      &mesg_attr);
-		if (ret < 0) {
-			LOG_ERR("Failed to query SCMI system capabilities: %d", ret);
-			return ret;
-		}
-
-		if (!(mesg_attr & SCMI_SYSTEM_MSG_ATTR_WARM_RESET)) {
-			LOG_WRN("Warm reset not supported by platform");
-			return -ENOTSUP;
-		}
-
 		cfg.system_state = SCMI_SYSTEM_POWER_STATE_WARM_RESET;
 		break;
 
@@ -47,7 +34,7 @@ static int scmi_reboot_handler(int type)
 
 	ret = scmi_system_power_state_set(&cfg);
 	if (ret < 0) {
-		LOG_ERR("System reboot failed with error: %d", ret);
+		LOG_ERR("Failed to request power state 0x%x: %d", cfg.system_state, ret);
 	}
 
 	return ret;

